Spigot-Server-Patches/0741-Improve-fixTileEntity.patch

```diff
@@ -0,0 +1,61 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Shane Freeder <theboyetronic@gmail.com>
+Date: Sun, 23 May 2021 17:33:28 +0100
+Subject: [PATCH] Improve fixTileEntity
+
+Improves fixTileEntity to actually remove invalid TEs from the world
+
+Also, maybe marginal pointless improvements to removeTileEntity to
+make it safer to call in more places without potentially busting state
+
+diff --git a/src/main/java/net/minecraft/server/level/WorldServer.java b/src/main/java/net/minecraft/server/level/WorldServer.java
+index 6aace2155258d8257d53ebfcb20c37ea7497c02c..ab4bf94233d1b50d74b3a6a77c92d64812c42472 100644
+--- a/src/main/java/net/minecraft/server/level/WorldServer.java
++++ b/src/main/java/net/minecraft/server/level/WorldServer.java
+@@ -407,11 +407,13 @@ public class WorldServer extends World implements GeneratorAccessSeed {
+             if (replacement != null) {
+                 replacement.setLocation(this, pos);
+                 this.setTileEntity(pos, replacement);
++        // Paper start
++                return replacement; // Paper
+             }
+-            return replacement;
+-        } else {
+-            return found;
+         }
++        this.removeTileEntity(pos);
++        return null;
```

Comment:
Perhaps we should add the fact it will return null to the log so we know when looking through logs? (Though it may be obvious from `type` not being an instance of `ITileEntity`, it would be much easier for us to debug if any issue arises)

Reply:
Yea, maybe it should log that it's actually removing the TE, that way you know, generally the way this works is pretty meh, but, vanilla generally seems to actually deal with getTileEntity returning null pretty well

Reply:
Would suck if for whatever reason, it nukes a legitimate tile entity, though. A change to the log is a tiny but appreciated change, I would argue.